.github/workflows/main.yml

```diff
@@ -71,6 +71,8 @@ jobs:
     timeout-minutes: 75
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
In the absence of the commit/tag history it is not possible to infer the version correctly. So we need a deep checkout.

This has been discussed in https://github.com/pypa/setuptools/pull/4537/files#r1701658826.

The approach seems to be the standard workaround for the same problem in `setuptools-scm`, see https://github.com/pypa/setuptools-scm/issues/480, https://github.com/pypa/setuptools-scm/issues/952#issuecomment-1751880320. Tracked in https://github.com/actions/checkout/issues/249 and https://github.com/actions/checkout/issues/1471.